modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/twostep/GridReduceQueryExecutor.java

```diff
@@ -385,6 +385,13 @@ else if (msg.page() == 0) // Count down only on each first page received.
         ReduceQueryRun lastRun = null;
 
         for (int attempt = 0;; attempt++) {
+            try {
+                cancel.checkCancelled();
+            }
+            catch (IgniteCheckedException cancelEx) {
```

Comment:
Can we declare `catch QueryCancelledException` here?

Reply:
done.